packages/fixie/src/agent.ts

```diff
@@ -317,30 +345,213 @@ export class FixieAgent {
         config.public
       );
     }
+    return agent;
+  }
+
+  static spawnAgentProcess(agentPath: string, port: number): ChildProcess {
+    term(`🌱 Starting local agent process on port ${port}...\n`);
+    const pathToCheck = path.resolve(path.join(agentPath, 'dist', 'index.js'));
+    if (!fs.existsSync(pathToCheck)) {
+      throw Error(`Your agent was not found at ${pathToCheck}. Do you need to build your agent code first?`);
+    }
+
+    // Ensure we kill child process on exit.
+    const controller = new AbortController();
+    const { signal } = controller;
+    process.on('exit', () => {
+      console.log('🌱 Terminating agent process');
+      controller.abort();
+    });
+
+    const cmdline = `npx --package=@fixieai/sdk fixie-serve-bin --packagePath ./dist/index.js --port ${port}`;
+    // Split cmdline into the first value (argv0) and a list of arguments separated by spaces.
+    term('🌱 Running: ').green(cmdline)('\n');
+    const [argv0, ...args] = cmdline.split(' ');
+    const subProcess = spawn(argv0, args, { cwd: agentPath, signal });
+    subProcess.stdout.on('data', (sdata: string) => {
+      console.log(`🌱 Agent stdout: ${sdata}`);
+    });
+    subProcess.stderr.on('data', (sdata: string) => {
+      console.error(`🌱 Agent stdout: ${sdata}`);
+    });
+    subProcess.on('close', (returnCode: number) => {
+      term(`🌱 Agent child process exited with code ${returnCode} - `).red('exiting.\n');
```

Comment:
It seems like this is always a problem, right? So maybe we should have more user-facing error messaging here?